Directory.Packages.props

```diff
@@ -1,37 +1,37 @@
 <Project>
   <ItemGroup>
-    <PackageVersion Include="Autofac.Extensions.DependencyInjection" Version="4.2.2" />
+    <PackageVersion Include="Autofac.Extensions.DependencyInjection" Version="10.0.0" />
```

Comment:
The main reason I hadn't previously updated this was because the version jump was so large I didn't want to break things and I was unsure whether the existing tests would find any breakage - have you checked this works?

Reply:
`Autofac.Extensions.DependencyInjection` is used only in one project with tests:

![image](https://github.com/user-attachments/assets/21818960-0fa6-4a92-a78b-05e95d15ca4c)

Reply:
> I was unsure whether the existing tests would find any breakage

Sure, but I don't know if our tests _actually_ cover its usage - if they don't, then them being green is meaningless. Could you look into it and determine whether or not what we already have is "good enough" to be confident in the upgrade?

I joined the project this year, so I have no prior knowledge to answer this question.

Reply:
This `CliExampleWithFactory` project is not working on `master` - it has 4 `DependencyResolutionException` exceptions at startup. This is why it is not included in any tests.

I reverted this Autofac update and this should be fixed in separate PR together with new integration tests